Formula/tektoncd-cli.rb

```diff
@@ -0,0 +1,24 @@
+class TektoncdCli < Formula
+  desc "CLI for interacting with TektonCD"
+  homepage "https://github.com/tektoncd/cli"
+  url "https://github.com/tektoncd/cli/archive/v0.3.0.tar.gz"
+
+  sha256 "7dda103e335ee5f5d696d6c61dbe786bd21c8a357948d965b301a120e20c3e58"
+
+  depends_on "go" => :build
+
+  def install
+    system "make", "bin/tkn"
+
+    bin.install "bin/tkn" => "tkn"
+    output = Utils.popen_read("SHELL=bash #{bin}/tkn completion bash")
+    (bash_completion/"tkn").write output
+    output = Utils.popen_read("SHELL=zsh #{bin}/tkn completion zsh")
+    (zsh_completion/"_tkn").write output
+    prefix.install_metafiles
+  end
+
+  test do
+    system "#{bin}/tkn", "version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
@SMillerDev sounds good thanks for the detailed description, i have updated the PR,